src/index.ts

```diff
@@ -180,6 +187,38 @@ export default class Position {
 		// pick the best!
 		return weightedPlacements[0].placement;
 	}
+	
+	findBestPlacementAt(offset: Offset, reference: Element, target: Element, placements: string[]) {
```

Comment:
So we're going to want to keep this and `findBestPlacement` consistent ... I think the addition of the `container` (`reference` here) is great, so that's something we'll want to make use of...

Rough function signatures:
```typescript
findBestPlacement(
  reference: Element, 
  target: Element, 
  placements: string[], 
  // little more typing, but using a function means we can just compute a generic ReferenceRect
  containerFunction: () => ReferenceRect = this.defaultContainerFunction,
  positionFunction = this.findPosition)

findBestPlacementAt(
  offset: Offset, 
  target: Element, 
  placements: string[], 
  containerFunction: () => ReferenceRect = this.defaultContainerFunction)

// with the rest of the protected methods
protected defaultContainerFunction(): ReferenceRect {
 return {
    // we go with window here, because that's going to be the simple/common case
    width: window.innerHeight - window.scrollY,
    height: window.innerWidth - window.scrollX
  };
}
```
Since `containerFunction` and `positionFunction` have default values it keeps the common case simple (just 3 arguments, and it mostly matches the existing `findX` methods)

The first few lines of `findBestPlacement` would change to: 
```typescript
// ...
const weightedPlacements = placements.map(placement => {
  const pos = positionFunction(reference, target, placement);
  let box = this.getPlacementBox((target as HTMLElement), pos);
  let hiddenHeight = box.bottom - containerFunction().height;
  let hiddenWidth = box.right - containerFunction().width;
// ...
```


Rough `findBestPlacementAt` implementation:
```typescript
const positionAt = (_, target, placement) => {
  this.findPositionAt(offset, target, placement);
};

`return findBestPlacement(reference, target, placements, positionAt);`
```

Reply:
Some of this I understand, however, if we do:
```js
  let hiddenHeight = box.bottom - containerFunction().height;
  let hiddenWidth = box.right - containerFunction().width;
```

That means one would have to call `findBestPlacementAt` with:
```js
findBestPlacementAt(
  offset: { top: 300, left: 500}, 
  target: document.querySelector("div.1"), 
  placements: ["left", "right"], 
  containerFunction: (ref) => {
    return {
      width: ref. offsetWidth,
	  height: ref. offsetHeight
    }
  }
```

Seems like we just need some logic inside of `findBestPlacement` to take care of our `containerFunction` problem here